diff --git a/protos/Android.bp b/protos/Android.bp
index eed6fd361..929925ac3 100644
--- a/protos/Android.bp
+++ b/protos/Android.bp
@@ -20,8 +20,16 @@ rust_library {
     cargo_pkg_version: "0.1.0",
     crate_root: "src/lib.rs",
     edition: "2021",
-    rustlibs: ["libprotobuf"],
+    features: [
+        "composite-disk",
+        "registered_events",
+    ],
+    rustlibs: [
+        "libcdisk_spec_proto",
+        "libprotobuf",
+        "libregistered_events_proto",
+    ],
     apex_available: ["com.android.virt"],
 }
 
 rust_protobuf {
